cmd/terramate/cli/cli.go

```diff
@@ -466,26 +467,25 @@ func (c *cli) printMetadata() error {
 	return nil
 }
 
-func (c *cli) runOnStacks(basedir string) error {
-	if !filepath.IsAbs(basedir) {
-		basedir = filepath.Join(c.wd, basedir)
-	}
-
-	mgr := terramate.NewManager(basedir, c.baseRef)
-	entries, err := c.listStacks(basedir, mgr, c.parsedArgs.Run.Changed)
+func (c *cli) runOnStacks() error {
+	mgr := terramate.NewManager(c.root(), c.prj.baseRef)
+	entries, err := c.listStacks(mgr, c.parsedArgs.Changed)
 	if err != nil {
 		return err
 	}
 
-	if c.parsedArgs.Run.Changed {
+	if c.parsedArgs.Changed {
 		c.log("Running on changed stacks:")
 	} else {
 		c.log("Running on all stacks:")
 	}
 
-	stacks := make([]stack.S, len(entries))
-	for i, e := range entries {
-		stacks[i] = e.Stack
+	relwd := prj.RelPath(c.root(), c.wd())
+	stacks := make([]stack.S, 0, len(entries))
+	for _, e := range entries {
+		if strings.HasPrefix(e.Stack.Dir, relwd) {
```

Comment:
This kind of filter is done at least on printRunOrder too, I think, I also think it is some kind of filtering going on, maybe there is some way to make it more explicit.